app/src/lib/git/branch.ts

```diff
@@ -39,7 +39,14 @@ export async function deleteBranch(repository: Repository, branch: Branch, accou
 
   // If the user is not authenticated, the push is going to fail
   // Let this propagate and leave it to the caller to handle
-  if (remote) {
+  if (!remote) {
+    return true
+  }
+
+  const isBranchExistsOnRemote = await checkIfBranchExistsOnRemote(repository, branch, account, remote)
```

Comment:
Is it acceptable variable and local function names?
Didn't see such long ones in your codebase yet.